docs/DeveloperGuide.md

```diff
@@ -325,92 +325,179 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+  * 2b1. SellSavvy gives a warning that there are similar tags in the new customer.
+  
+  Use case resumes from step 3.
+
 **Use case 3: Delete Customer and All Orders Related to The Customer**
 
 * **System**: SellSavvy
 * **Actor**: User
 * **Use Case**: UC03 - Delete Customer and All Orders related to the customer
+* **Preconditions**: There are customers displayed in the customer list.
 * **Guarantees**:
     * Customer and all their orders will be deleted if input parameters are valid.
 
 **MSS**
 
-1.  User wants to delete a customer.
-2.  User chooses to <ins>view all customers as a list (UC01)</ins>.
-3.  SellSavvy displays the list of customers.
-4.  User finds the customer index from the list.
-5.  User deletes the customer by his index.
-6.  SellSavvy updates the list of customers and indicates that delete is successful.
+1. User wants to delete a customer.
+2. User finds the customer index from the list.
+3. User deletes the customer by their index.
+4. SellSavvy updates the displayed list of customers and indicates that delete is successful.
 
 Use case ends.
 
-**Use case 4: List Orders by Customer**
+**Extensions**
+
+* 3a. SellSavvy detects that there are no customers with the specified index.
+  * 3a1. SellSavvy displays an error that the customer index is invalid. 
+  
+  Use case ends.
+
+**Use case 4: Find the Customer by their Name**
 
 * **System**: SellSavvy
 * **Actor**: User
-* **Use Case**: UC04 - List Orders by Customer
+* **Use Case**: UC04 - Find the Customer by their Name
 * **Guarantees**:
-    * Orders made by specific customer will be displayed as a list, if input parameters are valid.
+  * All customers displayed will have at least one of the specified keywords in their name.
+
+**MSS**
+
+1. User wants to search for a specific customer.
+2. User specifies keyword(s) of the name of the customer
+3. SellSavvy displays all customers whose names have at least one of the keywords.
+
+Use case ends.
+
+**Use case 5: Add an Order under a Customer**
+
+* **System**: SellSavvy
+* **Actor**: User
+* **Use Case**: UC06 - Add an Order under a Customer
+* **Preconditions**: There are customers displayed in the customer list.
+* **Guarantees**:
+    * A new pending order will be added under the specified customer, if input parameters are valid.
 
 **MSS**
 
-1.  User wants to view all orders made by a specific customer
-2.  User chooses to <ins>view all customers as a list (UC01)</ins>.
-3.  User inputs command to list all orders, by the index of customer in contact list.
-4.  SellSavvy validates the input, and retrieves a list of all orders made by specified customer.
-5.  SellSavvy displays the orders in a list in GUI.
+1. User wants to add an order under a specific customer.
+2. User finds the customer index from the list.
+3. User adds the customer using the index, specifying the details of the order.
+4. SellSavvy adds the order under the customer in the database.
```

Comment:
Not sure what you mean by database, u mean json data or add to customer in customer list? Database is a bir weird.